WebContent/zip-ext.js

```diff
@@ -208,11 +208,165 @@
 	FileWriter.prototype = new Writer();
 	FileWriter.prototype.constructor = FileWriter;
 
+	function DBWriter(contentType) {
+		var db, tempDB, that = this, blobs, dbName = "zipjs", instance;
+
+		function init(callback, onerror) {
+			var request = indexedDB.open(dbName, 5);
+			request.onerror = onerror;
+			request.onupgradeneeded = function (event) {
+				db = event.target.result;
+				db.createObjectStore("instances", { autoIncrement: true });
+			};
+			request.onsuccess = function (event) {
+				db = event.target.result;
+				addInstance(callback, onerror);
+			};
+			blobs = [];
+		}
+
+		function addInstance(callback, onerror) {
+			var t = db.transaction(["instances"], "readwrite"),
+				request = t.objectStore("instances").put(Date.now());
+			request.onerror = onerror;
+			request.onsuccess = function (event) {
+				instance = request.result;
+				cleanUpOldInstances();
+				window.addEventListener("storage", pongDB);
+				window.addEventListener("unload", close);
+				addInstanceDB(callback, onerror);
+			};
+		}
+
+		function addInstanceDB(callback, onerror) {
+			var request;
+			try {
+				request = indexedDB.open(dbName + "_" + instance, {temporary: true});
+			} catch (e) {
+				request = indexedDB.open(dbName + "_" + instance);
+			}
+			request.onerror = onerror;
+			request.onupgradeneeded = function (event) {
+				tempDB = event.target.result;
+				tempDB.createObjectStore("files", { autoIncrement: true });
+			};
+			request.onsuccess = function (event) {
+				callback();
+			};
+		}
+
+		function close() {
+			blobs = [];
+			indexedDB.deleteDatabase(dbName + "_" + instance);
+			window.removeEventListener("storage", pongDB);
+			window.removeEventListener("unload", pongDB);
+		}
+
+		function broadcastPingDB() {
+			sessionStorage.zipjs = Date.now();
+		}
+
+		function pongDB(event) {
+			if ((event !== undefined && event.key === "zipjs") && instance) {
+				db.transaction(["instances"], "readwrite")
+								.objectStore("instances")
+								.put(Date.now(), instance);
+			}
+		}
+
+		function cleanUpOldInstances() {
+			broadcastPingDB();
+			pongDB();
+			setTimeout(function () {
+				findOldInstances(10000);
+			}, 5000);
```

Comment:
Please avoid magic numbers without documentation. Explain why you use a timer, and why the value is 5 seconds, so that future users/contributors know why the code is written in the way it is.